### ISSUE TITLE
Choose and install an audio player

### ISSUE THREAD

**Issue by jensimmons** [jensimmons/thewebahead#8]
We want a pretty HTML5-based audio player. That has some cool features for playing files. That's easy to use & works everyplace. (Maybe also has a flash fallback, but increasingly that's not a priority. It's just IE6-8 and Opera Mini who can't play HTML5 audio.) 

NOTE:
All the sound files are coming from a big, official CDN Stats system. Those files have URLs like: http://fdlyr.co/d/webahead/cdn.5by5.tv/audio/broadcasts/webahead/2014/webahead-065.mp3

**Whatever audio player solution we implement _must_ play to those files, from the fdlyr.co URL.** 

I've also built a field for every episode to have an attached audio file, which will get stored in the Drupal system. Those should NOT be played by the player. Those are just for archival nuclear winter use cases. The plan is for those files to just sit there and never ever get used. VERY IMPORTANT. MUST POINT TO FEEDLAYER URL.

Implementing a player might be super easy. We might just want to do this in a custom node tpl for the episode type. Or perhaps a module with a library is a better solution. I'm not sure. Research is the first step. Pros? Cons? Why do which method? The fact we are pointing to a URL that's actually a redirect to another URL on a CDN system might make this complicated, or it might be no big deal. Don't know yet. 

I'd like to have a player that offers embed code to people, so they can easily embed that same player on their blog. 5by5 has this now! (Did you ever notice? Cool, eh?) 

![5by5 the web ahead 78 offline with john allsopp 2014-08-06 19-00-13 2014-08-06 19-00-16](https://cloud.githubusercontent.com/assets/108474/3835340/786e1b74-1dbd-11e4-98df-ea9225ecfa92.jpg)

**Comment by PhilJinAus** [comment]
Could we use a polyfill in IE8 (conditionally loaded - audioJS, perhaps)? Cautious of throwing JS at a problem generally speaking, but might be the best way to ensure cross-browser issues minimized...

In terms of a module based solution, Media Mogul could be an answer (and comes prepackaged with a Flash-based fallback...), but I haven't played with it, will investigate - https://www.drupal.org/project/mogul - only in dev, however and hasn't been updated for a couple of years so maybe not... building something bespoke may be the solution here.

**Comment by JasonNeel** [comment]
I think it would be cool to do something with the audio player similar to how ShopTalk Show does with theirs: having the ability to jump to a specific point in the show with a link. You can see this on their show notes for a particular show, where they deep link to the point where a particular question is raised. 

Would that be appropriate for this type of podcast? I know ShopTalk Show is more Q&A-based, whereas The Web Ahead is much more conversational. Maybe instead think of how CreativeMornings has their quote functionality? (http://creativemornings.com/quotes) I think that would be really neat thing to see with this site.

**Comment by mattsugihara** [comment]
FWIW, I'd find it super helpful. It'd
be great to be able to jump to like, when someone begins speaking, or as
you said Jen, to a particular quote.

On Saturday, August 9, 2014, Jason Neel notifications@github.com wrote:

> I think it would be cool to do something with the audio player similar to
> how ShopTalk Show does with theirs: having the ability to jump to a
> specific point in the show with a link. You can see this on their show
> notes for a particular show, where they deep link to the point where a
> particular question is raised.
> 
> Would that be appropriate for this type of podcast? I know ShopTalk Show
> is more Q&A-based, whereas The Web Ahead is much more conversational. Maybe
> instead think of how CreativeMornings has their quote functionality? (
> http://creativemornings.com/quotes) I think that would be really neat
> thing to see with this site.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/jensimmons/thewebahead/issues/8#issuecomment-51691522
> .

## 

## 

Matt Sugihara

**Comment by jensimmons** [comment]
I'm tempted to blog off IE8. And go with 100% HTML... why have a complex plugin library? Ohhhhh, right. Firefox doesn't do mp3. The codec wars aren't over yet. Flash fallback is needed. Plus extra JavaScripty awesomeness is always cool. 

Debated mediaelement vs jPlayer. Mediaelement looks more popular with the cool kids. jPlayer looks to be better documented. Both are being used about equally by Drupal sites (at about 9,000+ each). jPlayer module is better maintained. Mediaelement was written by some Drupal dev powerhouses. I've uploaded them both to try out. Everything else out there looks not as good.

I'm leaning towards mediaelement. That's what 5b5 is currently using. And what Shoptalk is using too — with those cool links that jump into the show. 

I _do_ want to pull a quote from every episode and highlight it. (I've already added the field to the episode content type. Did I add a ticket yet about choosing those quotes? No... yeah, other things are a higher priority). I love the idea of linking those quotes to the audio file, click and hear that quote in particular. Seems like work to find other places in the show to post shortcut links to that audio, but it also seems like it'd be really cool and useful. Maybe another place long term where Team The Web Ahead can help. Meanwhile, though, Focus On Minimum Shippable. AKA, the simple audio player.
